### PR TITLE
Add fakejoin.KubernetesSigner test helper

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR:
- exposes the claims structs from `lib/kubernetestoken` so they can be reused by other packages
- adds a fakejoin package containing helpers to use delegated join methods in tests
- adds `fakejoin.KubernetesSigner` that can sign Kubernetes-like JWTs allowing to use the kube jwks join method in tests

### CODE REVIEW

File: integrations/lib/testing/fakejoin/kubesigner.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+// SignServiceAccountJWT returns a signed JWT valid 30 minutes (1 min in the past, 29 in the future).
+// This token has the Teleport cluster name in its audience as required by the Kubernetes JWKS join method.
+func (s *KubernetesSigner) SignServiceAccountJWT(pod, namespace, serviceAccount, clusterName string) (string, error) {
+	now := s.clock.Now()
+	claims := kubernetestoken.ServiceAccountClaims{
+		Claims: jwt.Claims{
+			Subject:  fmt.Sprintf("system:serviceaccount:%s:%s", namespace, serviceAccount),
+			Audience: jwt.Audience{clusterName},
+			// To protect against time skep issues, we sign 1 min in the past.
```
Comment (marcoandredinis) [inline integrations/lib/testing/fakejoin/kubesigner.go:97]:
```suggestion
			// To protect against time skew issues, we sign 1 min in the past.
```

**Review by marcoandredinis — APPROVED** [review]

**Review by strideynet — APPROVED** [review]
🥳  I can see this becoming very useful, thank you.

**Comment by hugoShaka** [comment]
> Is it worth moving the package to kube/token?

I'd like to keep this code out of production packages as this is only for tests. I can create `lib/kubetoken/testutils` or something if we want to use this helper package in Teleport's code.

For now I don't think we plan to do this so I'd rather wait for us to need the helper in something which is not an integration before moving it in `lib`.

**Comment by tigrato** [comment]
> > Is it worth moving the package to kube/token?
> 
> I'd like to keep this code out of packages containing real code as this is only for tests. I can create `lib/kubetoken/testutils` or something if we want to use this helper package in Teleport's code.
> 
> For now I don't think we plan to do this so I'd rather wait for us to need the helper in something which is not an integration before moving it in `lib`.

I mean, move the whole kubernetestoken package to kube/token. This was always confusing to me since we are related to the same exact thing and we have a dedicated package for kube

**Comment by hugoShaka** [comment]
> I mean, move the whole kubernetestoken package to kube/token. This was always confusing to me since we are related to the same exact thing and we have a dedicated package for kube

Ah, I get it now. I agree this would make more sense. I'll move this around.

**Pull request merged**